src/libs/auth.js

```diff
@@ -134,6 +137,23 @@ authStore.subscribe((state, oldState) => {
   }
 })
 
+export const refreshTerraProfile = () => new Promise(async (resolve, reject) => {
+  try {
+    const profile = Utils.kvArrayToObject((await Ajax().User.profile.get()).keyValuePairs)
+    authStore.update(state => _.set('profile', profile, state))
+    resolve()
+  } catch (error) {
+    reject(error)
+  }
+})
+
+authStore.subscribe((state, oldState) => {
+  if ((!oldState.isSignedIn && state.isSignedIn) ||
+    (oldState.registrationStatus !== 'registered' && state.registrationStatus === 'registered')) {
```

Comment:
Because of the sequence of state transitions, this check will probably result in two profile loads on page load. I think it's sufficient just to look at `isSignedIn`